### PR TITLE
Add basic integration testing

### PR DESCRIPTION
**Pull request by rbradford**
All the fiddly cloud-init stuff has been worked out now and it is possible to get a very basic integration test.

Short term before merging:

- [x] Download the cloud image
- [x] Force single test execution due to hardcoded IP
- [x] Put under new feature to prevent Travis from executing it
- [x] Copy the source image every time
- [x] Extract reboot code
- [x] Add more tests (different number of CPUS/memory and also check entropy)
- [x] Integrate into CI

### CODE REVIEW

**Comment by rbradford** [comment]
Now that this is working and integrated into Jenkins we can have a think about reviewing this.

**Comment by sboeuf** [comment]
@rbradford yes I'll definitely have a look :)

**Comment by rbradford** [comment]
This is ready for integration.

@sameo @sboeuf i've intentionally kept the commit in with the original direct based approach even though I later switched to the binary executing based approach as a reference for the future.

File: src/main.rs [hunk 1]
```diff
@@ -116,3 +116,123 @@ fn main() {
...
+                        panic!("Took too many attempts to run command. Last error: {:?}", e);
+                    }
+                }
+            };
+            thread::sleep(std::time::Duration::new(10, 0));
+        }
+        s
+    }
+
+    fn prepare_files() -> (Vec<&'static str>, String) {
```
Comment (sameo) [inline src/main.rs]: Why do disks paths have to be static references?

Comment (sameo) [reply src/main.rs]: Nevermind:

```Rust
let osdisk_path = "/tmp/osdisk.img";
let cloudinit_path = "/tmp/cloudinit.img";

```

File: src/main.rs [hunk 2]
```diff
@@ -116,3 +116,123 @@ fn main() {
...
+                >= 1000
+        );
+
+        ssh_command("sudo reboot");
+
+        handler.join().unwrap();
+    }
+
+    #[test]
+    fn test_simple_launch_again() {
```
Comment (sameo) [inline src/main.rs]: Any reason why we're running that test twice?

Comment (rbradford) [reply src/main.rs]: This originally came from when the tests were being running via the internals and it was useful to check that running a second test worked (it didn't at some point.) With running via the binary the value is no longer there and I will remove it.

**Pull request merged**